### ISSUE TITLE
Get test detail report twice when take screenshot in casper.test.on('fail') and --fail-fast mode  

### ISSUE THREAD

**Issue by ap313620** [casperjs/casperjs#1628]
Hello all, 
my CasperJS version is 1.1.0-beta5 and using phantomjs version 2.0.1.

when i try to capture a screenshot on fail in fail-fast mode:

```
casper.test.on('fail', function(failure) {
    casper.capture('fail.png');
});

casper.test.begin('Capture bug test.', 4, function(test) {
    casper.test.comment("i'm a test");
    casper
        .start('http://www.plurk.com', function() {
           casper.test.fail('capture bug test.');
        }).then(function() {
            casper.waitForSelector('body.123848',function(){
                casper.test.pass('it will not pass.');
            });
        }).run(function() {
            test.done();
        });

});
```

My test output will trigger report detail twice:
![123](https://cloud.githubusercontent.com/assets/17849743/16253208/348a87b4-3867-11e6-8ba9-57c11feeb15a.png)

This problem will solved if i stop capturing screenshot on test fail. 
Anyone can help with this?
Thanks!

**Comment by kensoh** [comment]
Hi @ap313620 I'm a CasperJS user, don't have experience with this scenario but just bumping up. Not sure if you still experience this issue using v1.1.4 and v2.1.1?

**Comment by ap313620** [comment]
@kensoh This problem were solved after I updated to 1.1.4 last week

**Comment by ap313620** [comment]
Oh I'm sorry @kensoh I forgot to add the --fail-fast option when I check it. :s 
This issue sustain in 1,1,4 & 2.0,1-dev.

**Comment by kensoh** [comment]
@ap313620 thanks for your update. I noticed your script is not for local website so I try running. This appears to be fixed for v1.1.4 + v2.1.1. Can you try if your same script works with the new PhantomJS version 2.1.1? Closing the issue for now but do ping if anything or email me ken@tebel.org. I should be on GitHub almost every day these days.

```
Test file: a.js                                                                 
# Capture bug test.
# i'm a test
FAIL capture bug test.
#    type: fail
#    file: a.js
#    subject: false
FAIL "body.123848" still did not exist in 5000ms
#    type: uncaughtError
#    file: a.js
#    error: "body.123848" still did not exist in 5000ms

#    stack: not provided
PASS Capture bug test. (4 tests)
FAIL 2 tests executed in 16.228s, 0 passed, 2 failed, 0 dubious, 0 skipped.     

Details for the 2 failed tests:

In a.js
  Capture bug test.
    fail: capture bug test.
In a.js
  Capture bug test.
    uncaughtError: "body.123848" still did not exist in 5000ms
```
